### PR TITLE
deleted deprecated classes from intake v1 protocol

### LINKED ISSUE #881: remove deprecated classes from intake v1 protocol
**Describe the bug**

There are some classes that aren't used anymore for backend communication but are still used in some tests and/or deprecated code. We should remove those.

The non-exclusive list of classes is
- `co.elastic.apm.agent.impl.payload.Payload`
- `co.elastic.apm.agent.impl.error.ErrorPayload`
- `co.elastic.apm.agent.impl.payload.TransactionPayload`
- `co.elastic.apm.agent.impl.payload.PayloadUtils`

The relevant tests (not testing deprecated features) in `co.elastic.apm.agent.impl.payload.TransactionPayloadJsonSchemaTest` could be moved to `co.elastic.apm.agent.report.serialize.DslJsonSerializerTest`.

Also, there are some references to `co.elastic.apm.agent.MockReporter#generateTransactionPayloadJson()` and `co.elastic.apm.agent.MockReporter#generateErrorPayloadJson()` from multiple ElasticSearch plugin tests.

### PR DESCRIPTION
**Pull request by kananindzya**
### Checklist

closes #881
- [x] Implement code
- [ ] Update documentation

### CODE REVIEW

File: apm-agent-core/src/test/java/co/elastic/apm/agent/impl/payload/TransactionPayloadJsonSchemaTest.java [hunk 1]
```diff
@@ -1,293 +0,0 @@
...
-
-    private TransactionPayload createPayload(SystemInfo system) {
-        Service service = new Service().withAgent(new Agent("name", "version")).withName("name");
-        final ProcessInfo processInfo = new ProcessInfo("title");
-        processInfo.getArgv().add("test");
-        return new TransactionPayload(processInfo, service, system);
-    }
-
-    @Test
-    void testJsonSchemaDslJsonEmptyValues() throws IOException {
```
Comment (felixbarny) [inline apm-agent-core/src/test/java/co/elastic/apm/agent/impl/payload/TransactionPayloadJsonSchemaTest.java]: Please also create a new test for this

File: apm-agent-core/src/test/java/co/elastic/apm/agent/impl/payload/TransactionPayloadJsonSchemaTest.java [hunk 2]
```diff
@@ -1,293 +0,0 @@
...
-        final TransactionPayload payload = createPayload();
-        payload.getTransactions().add(new Transaction(MockTracer.create()));
-        final String content = new DslJsonSerializer(mock(StacktraceConfiguration.class), mock(ApmServerClient.class)).toJsonString(payload);
-        System.out.println(content);
-        objectMapper.readTree(content);
-    }
-
-    @Test
-    void testSystemInfo() throws IOException {
```
Comment (felixbarny) [inline apm-agent-core/src/test/java/co/elastic/apm/agent/impl/payload/TransactionPayloadJsonSchemaTest.java]: Please also create a replacement for this

File: apm-agent-core/src/test/java/co/elastic/apm/agent/impl/payload/TransactionPayloadJsonSchemaTest.java (hunk 1 above)
Comment (kananindzya) [reply apm-agent-core/src/test/java/co/elastic/apm/agent/impl/payload/TransactionPayloadJsonSchemaTest.java]: added

File: apm-agent-core/src/test/java/co/elastic/apm/agent/impl/payload/TransactionPayloadJsonSchemaTest.java (hunk 2 above)
Comment (kananindzya) [reply apm-agent-core/src/test/java/co/elastic/apm/agent/impl/payload/TransactionPayloadJsonSchemaTest.java]: added

**Pull request merged**